splunk_add_on_ucc_framework/ucc_ui_lib/src/main/webapp/components/CheckBoxComponent.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Switch from '@splunk/react-ui/Switch';
+
+class CheckBoxComponent extends Component {
+    constructor(props) {
+        super(props);
+    }
+
+    handleClick = (e) => {
+        this.props.handleClick(this.props.id, e.target.value);
```

Comment:
What would be the type of `e.target.value` here? , if it's 0/1 then it's fine. but if it is true/false then you need to convert before calling the handleClick as you're expecting 0/1 in `this.props.value` .  Also, rename the function to `this.props.handleChange`. because all the components will receive the same function.

Also, if possible then check what's expected value on REST Handler side for Radiobox. **true/false or 0/1** and call `this.props.handleChange` with that value ? . So later We won't need to parse value again before REST call.

Reply:
Updated funcation name to `this.props.handleChange` and it returns 0/1 value with field's id.
for **CheckBox** REST Handler uses 0/1 value. So, no need to parse the value.
for **RadioBar** return value would be item value provided in the GlobalConfig which will be used in REST Handler.